pkg/internal/kms/localkms/localkms_aead.go

```diff
@@ -0,0 +1,69 @@
+/*
```

Comment:
it will take more time to split into smaller commits.. I can probably separate KMS and SecretLock into 2 separate commits. Let me know if this helps...

internal package introduced in this PR (kms/localkms) is solely used by the default KMS implementation to :
1. support local read/write operations of keysets for the local kms (not needed for non-local KMSs)
2. offer a (local) KMS client (that takes a secret lock and returns an AEAD for key wrapping) to be registered by Tink.
3. separate specific 'internal' logic needed by localkms (and Tink) from the rest of the framework. This helps readability for the user.

This logic should not be exposed/referenced by the user in anyway, different KMS implementations will (will probably not need to reference this internal package) have their own logic to create KMS clients and key storage. For example GCP or AWS KMS integrations (with Tink) do not need to store/wrap keys locally, they use a (different) client connecting to the cloud to fetch keys with a SecretLock connecting to different backend. They might reuse the client creation logic however, it's probably best to rename this internal package as `kms/support`, I'll have a second look at how this internal package is structured ..

Look at `NewKMS(p provider)` function in  `pkg/kms/localkms/localkms.go` below  to see how the secretLock and this local kms client are created and registered in Tink.

Reply:
@Baha-sk 1700 LOC is difficult to review and iterate on. I would prefer to find some smaller commits so that reviewing updates to the PRs is easier.

Reply:
I still have additional changes to integrate this new KMS in `framework`, although they can be done in a separate PR..